rllib/policy/torch_policy.py

```diff
@@ -98,8 +99,10 @@ def __init__(self,
         """
         self.framework = "torch"
         super().__init__(observation_space, action_space, config)
-        self.device = (torch.device("cuda")
-                       if torch.cuda.is_available() else torch.device("cpu"))
+        if torch.cuda.is_available() and ray.get_gpu_ids():
```

Comment:
Added the check for actually assigned GPUs for this particular Worker (ray.get_gpu_ids()).

Reply:
Nice! Did you try `CUDA_VISIBLE_DEVICES=-1 rllib train ...`?  Did it end up with CPU?